modules/websocket_bridge/include/hephaestus/utils/ws_client.h

```diff
@@ -0,0 +1,60 @@
+#pragma once
+
+#include <chrono>
+#include <csignal>
+#include <cstdint>
+#include <map>
+#include <memory>
+#include <optional>
+#include <string>
+#include <vector>
+
+#include <foxglove/websocket/websocket_client.hpp>
+#include <foxglove/websocket/websocket_notls.hpp>
+#include <google/protobuf/message.h>
+#include <hephaestus/utils/ws_protocol.h>
+
+namespace heph::ws {
+
+using WsClientNoTls = foxglove::Client<foxglove::WebSocketNoTls>;
+
+struct ServiceCallState {
```

Comment:
It is a bit of a code smell if a struct has lots of functions. I would either transform this in a class or move the function to free functions, or to another class

Reply:
I don't think a class makes sense, I know clang-tidy will then hit me with: no public member variables, which means I add all these accessors, ultimately I end up with an extra 50 lines of code, less readibility for a questionable gain IMO.

I agree on making the functions that actually DO something free functions, because those indeed smell (receive*()). But the convenience accessors which are only there for readability and laziness I would prefer to keep.

Reply:
done